### PR TITLE
[sil-combine] Enable dead alloc_ref elim for ossa.

### PR DESCRIPTION
**Pull request by gottesmm**
It was just looking for an alloc_ref that only has dealloc_ref,
set_deallocating, fix_lifetime users.

### CODE REVIEW

File: lib/SILOptimizer/SILCombiner/SILCombinerMiscVisitors.cpp [hunk 1]
```diff
@@ -680,11 +680,6 @@ SILInstruction *SILCombiner::visitAllocStackInst(AllocStackInst *AS) {
 }
 
 SILInstruction *SILCombiner::visitAllocRefInst(AllocRefInst *AR) {
-  if (AR->getFunction()->hasOwnership())
-    return nullptr;
-
-  if (!AR)
```
Comment (gottesmm) [inline lib/SILOptimizer/SILCombiner/SILCombinerMiscVisitors.cpp:686]: I don't know why this check is here. Seems like a mistake. A visitor should never be passing in a null value.

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Review by meg-gupta — APPROVED** [review]
lgtm. If the plan is to have multiple test files for sil-combine, maybe a new folder like `test/SILOptimizer/SILCombine` will be useful in the future.

**Comment by gottesmm** [comment]
I was thinking about that. I want to make sure @atrick is cool with it first.

**Comment by atrick** [comment]
@gottesmm @meg-gupta fyi, it's fine to eliminate `fix_lifetime` when the object is *uniquely identified* and there are no other uses, which is indeed the case here for `alloc_ref`. There are other places where we delete "dead" `fix_lifetime` instructions that are not uniquely referenced, which is definitely incorrect.

I have no problem with the test subdirectories matching the source subdirectories.

**Comment by gottesmm** [comment]
Ok. SGTM! I was just porting. I am trying to slim down my large patch.

**Pull request merged**